pipeline/lab.py

```diff
@@ -0,0 +1,301 @@
+    rig_description : varchar(1024) 
+    """
+
+@schema
+class Species(dj.Lookup):
+    definition = """
+    species: varchar(24)
+    """
+    contents = zip(['Mus musculus'])
+
+
+@schema
+class AnimalStrain(dj.Lookup):
+    definition = """
+    animal_strain       : varchar(30)
+    """
+    contents = zip(['pl56', 'kj18'])
+
+
+@schema
+class AnimalSource(dj.Lookup):
+    definition = """
+    animal_source       : varchar(30)
+    """
+    contents = zip(['Jackson Labs', 'Allen Institute', 'Charles River', 'MMRRC', 'Taconic', 'Other'])
+
+
+@schema
+class ModifiedGene(dj.Manual):
+    definition = """
+    gene_modification   : varchar(60)
+    ---
+    gene_modification_description = ''         : varchar(256)
+    """
+
+
+@schema
+class Subject(dj.Manual):
+    definition = """
+    subject_id          : int   # institution 6 digit animal ID
+    ---
+    -> [nullable] Person        # person responsible for the animal
+    cage_number=null    : int   # institution 6 digit animal ID
+    date_of_birth=null  : date  # format: yyyy-mm-dd
+    sex                 : enum('M','F','Unknown')
+    -> Species
+    -> [nullable] AnimalSource  # where was the animal ordered from
+    """
+
+    class Strain(dj.Part):
+        definition = """
+        # Subject gene modifications
+        -> master
+        -> AnimalStrain
+        """
+
+    class GeneModification(dj.Part):
+        definition = """
+        # Subject gene modifications
+        -> master
+        -> ModifiedGene
+        ---
+        zygosity = 'Unknown' : enum('Het', 'Hom', 'Unknown')
+        type = 'Unknown'     : enum('Knock-in', 'Transgene', 'Unknown')
+        """
+
+
+@schema
+class CompleteGenotype(dj.Computed):
+    # should be computed
+    definition = """
+    -> Subject
+    ---
+    complete_genotype : varchar(1000)
+    """
+
+    def make(self, key):
+        pass
+
+
+@schema
+class WaterRestriction(dj.Manual):
+    definition = """
+    -> Subject
+    ---
+    water_restriction_number    : varchar(16)   # WR number
+    cage_number                 : int
+    wr_start_date               : date
+    wr_start_weight             : Decimal(6,3)
+    """
+
+
+@schema
+class VirusSource(dj.Lookup):
+    definition = """
+    virus_source   : varchar(60)
+    """
+    contents = zip(['Janelia core', 'UPenn', 'Addgene', 'UNC', 'Other'])
+
+
+@schema
+class Serotype(dj.Manual):
+    definition = """
+    serotype   : varchar(60)
+    """
+
+
+@schema
+class Virus(dj.Manual):
+    definition = """
+    virus_id : int unsigned
+    ---
+    -> VirusSource 
+    -> Serotype
+    -> Person
+    virus_name      : varchar(256)
+    titer           : Decimal(20,1) # 
+    order_date      : date
+    remarks         : varchar(256)
+    """
+
+    class Notes(dj.Part):
+        definition = """
+        # Notes for virus
+        -> Virus
+        note_id     : int
+        ---
+        note        : varchar(256)
+        """
+
```

Comment:
Looks like there is another Virus table in the schema `virus`. Delete this one?

Reply:
Also the `SurgeryVirusInjection`

Reply:
Those are from the MAP pipelines, perhaps not applicable here, I have removed